### ISSUE TITLE
V6 endpoint names can contain '@', but older version can't send message to them

### ISSUE THREAD

**Issue by bording** [Particular/NServiceBus#3906]
While testing the problem described in https://github.com/Particular/PlatformDevelopment/issues/870 I discovered that V6 can use '@' in the endpoint name, and this works just fine.

For example, I created an endpoint with the name "Responder@V6" and in RabbitMQ, I ended up with that exact name for a queue. When sending from another V6 endpoint, everything works fine and messages are delivered properly.

However, when I tried to send to my "Responder@V6" endpoint from a V5 endpoint, everything after the '@' got stripped away from the endpoint name, and the message failed looking for a nonexistent "Responder" endpoint.

I also tried creating a V5 endpoint with the name "Sender@V5" and it ended up creating queues named "Sender" instead.

Seeing how this only works for V6, should we instead check for '@' in the endpoint name and consider it invalid instead and throw?

**Comment by yvesgoeleven** [comment]
@bording the @ sign has a meaning, whatever is behind the @ sign should be interpretted as the physical location of the queue, aka a rabbit server/cluster/connectionstring. If the transport does not support multi cluster routing, then just ignore whatever is behind the @ sign I'd say

**Comment by mikeminutillo** [comment]
@yvesgoeleven given that the physical address quite often ends up being `logicalEndpointName@something` then it probably doesn't make sense for logical endpoint name to contain an @ symbol. 

i.e. this should probably throw as it's likely to cause problems somewhere:

```CSharp
var endpointConfiguration = new EndpointConfiguration("Responder@V6");
```

If you run that under MSMQ you'll get a physical address of `Responder@V6@MachineName`

 You can have one @ in a physical address. In a logical address it should be an error.

**Comment by yvesgoeleven** [comment]
@mikeminutillo agreed, that should not be allowed

**Comment by timbussmann** [comment]
:+1: on doing basic sanity checking of endpoint names. Are there other characters we need to protect logical endpoint names from?

**Comment by danielmarbach** [comment]
I agree we should prevent it. Can we add this to the routing issue and tackle it there?

**Comment by timbussmann** [comment]
> I agree we should prevent it. Can we add this to the routing issue and tackle it there?

tbh this issue is about EndpointConfiguration, not the routing API (did you mean this issue: https://github.com/Particular/NServiceBus/issues/3914 ?). I'd say this seems to be a maintainer prio thing?

**Comment by timbussmann** [comment]
fixed by #3974
